nesta/packages/mag/query_mag_sparql.py

```diff
@@ -309,6 +336,149 @@ def query_authors(ids=None, results_limit=None):
+        OPTIONAL {{?paperId datacite:doi ?paperDoi .}}
+        OPTIONAL {{?paperId dcterms:language ?paperLanguage .}}
+        OPTIONAL {{?paperId magp:bookTitle ?bookTitle .}}
+        ?paperId fabio:hasDiscipline ?fieldOfStudyId .
+
+        FILTER (?paperCreatedDate >= "{from_date}"^^xsd:date)
+        FILTER (?paperCitationCount >= {min_citations})
+        FILTER (?affiliationGridId = <http://www.grid.ac/institutes/{grid_id}>)
+    }}
+    '''
+    logging.debug(f"Querying MAG for institute {grid_id}")
+
+    for batch in (sparql_query(MAG_ENDPOINT,
+                               nbatch=batch_size,
+                               batch_limit=batch_limit,
+                               query=query.format(grid_id=grid_id,
+                                                  from_date=from_date,
+                                                  min_citations=min_citations))):
+        for row in batch:
+            # extract any of ids
+            yield {k: extract_entity_id(v) if k.endswith('Id') else v
+                   for k, v in row.items()}
+
+
+def count_papers(institutes, done_institutes, paper_ids, intermediate_file,
+                 save_every=1000000, limit=None):
+    """Count the number of papers produced from a list of supplied institues.
+
+    Args:
+        institutes(list): GRID IDs of the institutes to look up
+        done_institutes(list): GRID IDs of institues which have already been processed
+        paper_ids(list): MAG IDs of papers which have already been processed
+        intermediate_file(:obj:`boto3.resources.s3.object`): file to store processed
+        institutes and paper ids
+        save_every(int): saves to s3 each time this number of papers are found
+        limit(int): break at this number of processed records, for testing
+
+    Returns:
+        (int): total number of papers found
+    """
+    query = '''
+    PREFIX magp: <http://ma-graph.org/property/>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX dcterms: <http://purl.org/dc/terms/>
+
+    SELECT ?paperId
+
+    WHERE {{
+    ?affiliationId magp:grid <http://www.grid.ac/institutes/{}> .
+    ?authorId org:memberOf ?affiliationId .
+    ?paperId dcterms:creator ?authorId .
+    }}
+    '''
+    count = 0
+    for institute in institutes:
+        for batch in sparql_query(MAG_ENDPOINT, query.format(institute)):
+            for row in batch:
+                count += 1
+                logging.debug(row)
+                paper_ids.add(extract_entity_id(row['paperId']))
+                total = len(paper_ids)
+                if not total % 1000:
+                    logging.info(f"{total:,} EU papers found")
+
+                if not total % save_every:
+                    # write out to file on S3 for safekeeping
+                    logging.info(f"Writing out to file with {len(done_institutes)} institutes")
+                    intermediate_file.put(Body=json.dumps({'paper_ids': list(paper_ids),
```

Comment:
is this the offending line??

Reply:
no :( 
it's this function: https://github.com/nestauk/nesta/blob/eff161bcf4c74ea0ee478475e7e0d9e0a086b4d2/nesta/packages/mag/query_mag_sparql.py#L150-L165